File: app/backend/java/auth-service/src/main/java/com/srbms/authservice/controller/UserController.java
package com.srbms.authservice.controller;

import com.srbms.authservice.dto.UserProfileResponse;
import com.srbms.authservice.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/users")
public class UserController {

    @Autowired
    private UserService userService;

    @GetMapping("/profile")
    public ResponseEntity<?> getMyProfile(
            @RequestHeader(value = "X-User-Id", required = false) String userIdHeader,
            @RequestParam(value = "userId", required = false) Integer queryUserId) {
        
        Integer targetUserId = null;
        if (userIdHeader != null && !userIdHeader.isBlank()) {
            targetUserId = Integer.parseInt(userIdHeader);
        } else if (queryUserId != null) {
            targetUserId = queryUserId;
        }

        if (targetUserId == null) {
            return ResponseEntity.badRequest().body(Map.of("error", "Missing User ID in header or query parameter"));
        }

        try {
            UserProfileResponse profile = userService.getUserProfile(targetUserId);
            return ResponseEntity.ok(profile);
        } catch (RuntimeException e) {
            return ResponseEntity.badRequest().body(Map.of("error", e.getMessage()));
        }
    }

    @GetMapping("/{id}")
    public ResponseEntity<?> getUserById(@PathVariable("id") Integer id) {
        try {
            UserProfileResponse profile = userService.getUserProfile(id);
            return ResponseEntity.ok(profile);
        } catch (RuntimeException e) {
            return ResponseEntity.badRequest().body(Map.of("error", e.getMessage()));
        }
    }

    @GetMapping("/all")
    public ResponseEntity<List<UserProfileResponse>> getAllUsers() {
        return ResponseEntity.ok(userService.getAllUsers());
    }

    @PutMapping("/{id}/status")
    public ResponseEntity<?> updateUserStatus(@PathVariable("id") Integer id, @RequestBody Map<String, String> body) {
        try {
            String status = body.get("status");
            if (status == null || status.isBlank()) {
                return ResponseEntity.badRequest().body(Map.of("error", "Status field is required"));
            }
            UserProfileResponse updatedProfile = userService.updateUserStatus(id, status);
            return ResponseEntity.ok(updatedProfile);
        } catch (RuntimeException e) {
            return ResponseEntity.badRequest().body(Map.of("error", e.getMessage()));
        }
    }

    @PutMapping("/{id}")
    public ResponseEntity<?> updateUser(@PathVariable("id") Integer id, @RequestBody Map<String, Object> body) {
        try {
            UserProfileResponse updatedProfile = userService.updateUser(id, body);
            return ResponseEntity.ok(updatedProfile);
        } catch (RuntimeException e) {
            return ResponseEntity.badRequest().body(Map.of("error", e.getMessage()));
        }
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<?> deleteUser(@PathVariable("id") Integer id) {
        try {
            userService.deleteUser(id);
            return ResponseEntity.ok(Map.of("message", "User deleted successfully", "userId", id));
        } catch (RuntimeException e) {
            return ResponseEntity.badRequest().body(Map.of("error", e.getMessage()));
        }
    }
}

File: app/backend/.net/SmartRide.AuthService/Controllers/UserController.cs
using Microsoft.AspNetCore.Mvc;
using SmartRide.AuthService.Models.DTOs;
using SmartRide.AuthService.Services;

namespace SmartRide.AuthService.Controllers;

[ApiController]
[Route("api/users")]
public class UserController : ControllerBase
{
    private readonly IUserService _userService;

    public UserController(IUserService userService)
    {
        _userService = userService;
    }

    [HttpGet("profile")]
    public async Task<IActionResult> GetMyProfile(
        [FromHeader(Name = "X-User-Id")] string? userIdHeader,
        [FromQuery(Name = "userId")] int? queryUserId)
    {
        int? targetUserId = null;
        if (!string.IsNullOrWhiteSpace(userIdHeader) && int.TryParse(userIdHeader, out var parsedHeaderId))
        {
            targetUserId = parsedHeaderId;
        }
        else if (queryUserId.HasValue)
        {
            targetUserId = queryUserId.Value;
        }

        if (!targetUserId.HasValue)
        {
            return BadRequest(new { error = "Missing User ID in header or query parameter" });
        }

        try
        {
            var profile = await _userService.GetUserProfileAsync(targetUserId.Value);
            return Ok(profile);
        }
        catch (Exception ex)
        {
            return BadRequest(new { error = ex.Message });
        }
    }

    [HttpGet("{id}")]
    public async Task<IActionResult> GetUserById([FromRoute] int id)
    {
        try
        {
            var profile = await _userService.GetUserProfileAsync(id);
            return Ok(profile);
        }
        catch (Exception ex)
        {
            return BadRequest(new { error = ex.Message });
        }
    }

    [HttpGet("all")]
    public async Task<ActionResult<List<UserProfileResponse>>> GetAllUsers()
    {
        var users = await _userService.GetAllUsersAsync();
        return Ok(users);
    }

    [HttpPut("{id}/status")]
    public async Task<IActionResult> UpdateUserStatus([FromRoute] int id, [FromBody] Dictionary<string, string> body)
    {
        try
        {
            if (!body.TryGetValue("status", out var status) || string.IsNullOrWhiteSpace(status))
            {
                return BadRequest(new { error = "Status field is required" });
            }

            var updatedProfile = await _userService.UpdateUserStatusAsync(id, status);
            return Ok(updatedProfile);
        }
        catch (Exception ex)
        {
            return BadRequest(new { error = ex.Message });
        }
    }

    [HttpPut("{id}")]
    public async Task<IActionResult> UpdateUser([FromRoute] int id, [FromBody] Dictionary<string, object> body)
    {
        try
        {
            var updatedProfile = await _userService.UpdateUserAsync(id, body);
            return Ok(updatedProfile);
        }
        catch (Exception ex)
        {
            return BadRequest(new { error = ex.Message });
        }
    }

    [HttpDelete("{id}")]
    public async Task<IActionResult> DeleteUser([FromRoute] int id)
    {
        try
        {
            await _userService.DeleteUserAsync(id);
            return Ok(new { message = "User deleted successfully", userId = id });
        }
        catch (Exception ex)
        {
            return BadRequest(new { error = ex.Message });
        }
    }
}
